base/applications/taskmgr/procpage.c

```diff
@@ -931,3 +933,140 @@ int CALLBACK ProcessPageCompareFunc(LPARAM lParam1, LPARAM lParam2, LPARAM lPara
 }
+
+static DWORD _GetModuleFileNameExW(HANDLE hProcess, HMODULE hModule, LPWSTR lpFilename, DWORD nSize)
+{
+    typedef DWORD (WINAPI *PFN_GETMODULEFILENAMEEXW)(HANDLE, HMODULE, LPWSTR, DWORD);
+
+    HMODULE hLibrary;
+    PFN_GETMODULEFILENAMEEXW func;
+    DWORD dwRet = 0;
+
+    hLibrary = LoadLibraryW(L"psapi.dll");
+
+    if (hLibrary)
+    {
+        func = (PFN_GETMODULEFILENAMEEXW)GetProcAddress(hLibrary, "GetModuleFileNameExW");
+
+        if (func)
+        {
+            dwRet = func(hProcess, hModule, lpFilename, nSize);
+        }
+
+        FreeLibrary(hLibrary);
+    }
+
+    return dwRet;
+}
+
+static BOOL GetProcessExecutablePath(DWORD dwProcessId, LPWSTR lpExePath, DWORD dwLength)
+{
+    WCHAR *pszExePath;
+    HANDLE hProcess;
+    BOOL bSuccess = FALSE;
+
+    if (dwProcessId == 0)
+    {
+        return FALSE;
+    }
+    
+    pszExePath = HeapAlloc(GetProcessHeap(), 0, dwLength * sizeof(WCHAR));
+
+    if (!pszExePath)
+    {
+        return FALSE;
+    }
+    
+    /* PID = 4 or "System" */
+    if (dwProcessId == 4)
+    {
+        static const WCHAR szKernelPath[] = L"\\SystemRoot\\System32\\ntoskrnl.exe";
+
+        if (dwLength >= _countof(szKernelPath))
+        {
+            StringCchCopyW(pszExePath, dwLength, szKernelPath);
+
+            bSuccess = TRUE;
+        }
+    }
+    else
+    {
+        hProcess = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, FALSE, dwProcessId);
+
+        if (hProcess)
+        {
+            bSuccess = (_GetModuleFileNameExW(hProcess, NULL, pszExePath, dwLength) != 0);
+
+            CloseHandle(hProcess);
+        }
+    }
+
+    if (bSuccess)
+    {
+        /* Remove or replace path prefix if present */
+        if (wcsncmp(pszExePath, L"\\??\\", 4) == 0)
+        {
+            /* Remove "\??\" prefix */
+            StringCchCopyW(pszExePath, dwLength, pszExePath + 4);
+        }
+        else if (_wcsnicmp(pszExePath, L"\\SystemRoot", 11) == 0)
+        {
+            WCHAR szWinDir[MAX_PATH];
+
+            bSuccess = (GetWindowsDirectoryW(szWinDir, _countof(szWinDir)) != 0);
+
+            if (bSuccess)
+            {
+                if (dwLength >= wcslen(szWinDir) + (wcslen(pszExePath) - 11) + 1)
+                {
+                    /* Replace "\SystemRoot" prefix */
+                    StringCchPrintfW(pszExePath, dwLength, L"%s%s", szWinDir, pszExePath + 11);
+                }
+                else
+                {
+                    bSuccess = FALSE;
+                }
+            }
+        }
+
+        if (bSuccess)
+        {
+            StringCchCopyW(lpExePath, dwLength, pszExePath);
+        }
+    }
+
+    HeapFree(GetProcessHeap(), 0, pszExePath);
+
+    return bSuccess;
+}
+
+void ProcessPage_OnProperties(void)
+{
+    DWORD dwProcessId;
+    WCHAR szExePath[MAX_PATH];
+
+    dwProcessId = GetSelectedProcessId();
+
+    if (GetProcessExecutablePath(dwProcessId, szExePath, _countof(szExePath)))
+    {
+        return;
+    }
+}
```

Comment:
unfinished still. I guess it's WIP.

Reply:
I tried both `ShellExecute` method but still not working.

Reply:
Is it only non-working in ros? Or is it also non-working in Windows? If one of the solutions would work on Windows, then it would be fine to use that. And let (someone else) fix the remaining bug otherwhere in ros later.

Reply:
Not tried it on Windows yet. I'll try it and use a working solution.

Reply:
Solution 1 works, tested on WinXP SP3